python/cudf_polars/cudf_polars/experimental/shuffle.py

```diff
@@ -192,9 +233,47 @@ def _(
 def _(
     ir: Shuffle, partition_info: MutableMapping[IR, PartitionInfo]
 ) -> MutableMapping[Any, Any]:
-    # Use a simple all-to-all shuffle graph.
+    # Extract "shuffle_method" configuration
+    if (
+        shuffle_method := ir.config_options.get(
+            "executor_options.shuffle_method",
+            default=None,
+        )
+    ) not in (*_SHUFFLE_METHODS, None):
+        raise ValueError(
+            f"{shuffle_method} is not a supported shuffle method. "
+            f"Expected one of: {_SHUFFLE_METHODS}."
+        )
+
+    # Try using rapidsmp shuffler if we have "simple" shuffle
+    # keys, and the "shuffle_method" config is set to "rapidsmp"
+    _keys: list[Col]
+    if shuffle_method in (None, "rapidsmp") and len(
+        _keys := [ne.value for ne in ir.keys if isinstance(ne.value, Col)]
+    ) == len(ir.keys):  # pragma: no cover
```

Comment:
I'm still parsing this `if` condition, but just to confirm: there are cases where the user sets `shuffle_method="rapidsmp"`, but we'll not use it? IIUC, that can happen when... some of the keys aren't plain `Col`umns? Not sure yet...

Reply:
Yes. I'm not sure how I want to approach the various cases here. We can only use the `rapidsmp` shuffle if (1) there is a compatible cluster running and (2) we are grouping on column names.

I'd like for us to **try** using the rapidsmp shuffler by default (since it's so much better than tasks). However, I don't think we should raise an error when we are grouping on complex expressions, or if the user didn't specify `"rapidsmp"` in their configuration. WDYT?

Reply:
> I don't think we should raise an error when we are grouping on complex expressions

Is there a path to supporting that in `rapidsmp`? If so, then I'd prefer to raise until that's implemented.

But I don't have a strong / well-informed opinion here.

Aside from this case, I like the rule of "use rapidsmp by default if it's available and fallback if it's unavailable. But raise if it's explicitly requested and available"

Reply:
>Is there a path to supporting that in rapidsmp?

I think so, but I'm not sure if it's common enough to be a high priority.

>I'd prefer to raise until that's implemented.

If 1/4 shuffles in a query is not supported, does it make sense to raise an error? Maybe the answer is "yes"? Maybe a warning is better?

Reply:
Oh, I didn't consider the multi-shuffle case at all.

In that case, I think that was you have is appropriate. No need for an error / warning by default when we fallback (but probably keep the error when `shuffle_method` is explicitly `"rapidsmp"`. Then we have an easy way of testing).